tests/codex/sales/testsales.nim

```diff
@@ -296,7 +296,7 @@ asyncchecksuite "Sales":
                          slot: UInt256,
                          onBatch: BatchProc): Future[?!void] {.async.} =
       let blk = bt.Block.new( @[1.byte] ).get
-      onBatch( blk.repeat(request.ask.slotSize.truncate(int)) )
+      discard await onBatch( blk.repeat(request.ask.slotSize.truncate(int)) )
```

Comment:
kinda weird that we're discarding the result here, why not handle/log it?

Reply:
OK so a bit of context. This code is not even supposed to compile, as `onBatch` returns a `Future` and this is neither used nor discarded. The older versions of Nim seemed to let that pass somehow, but the newer ones will scream and refuse to compile. I then try to infer what this is supposed to do - since no-one seemed to care about the result of `onBatch` on this test, I simply `await discard` it.

Reply:
Btw the reason I `await`ed this one instead of doing an `asyncSpawn` is because I had a similar conversation with @emizzle in the past and he advised me to use `await`. So for marketplace tests that's my default. :-)

Reply:
The fact that it wasn't being detected in previous versions is definitely a Nim/chronos bug. But what I'm sayin here is that we should do proper error handling in case this is returning a `Result`, not simply discarding it. At the very least, the error should be logged.

And I do understand that there are a few of this weird cases or code that was broken/buggy in the original implementation, but we should try to fix this as much as possible as soon as we catch them, larger efforts should be captured as issues as well so that they don't slip through the cracks.

Reply:
OK, in this particular case: since `onBatch` returns `Future[?!void]` as its last operation, I'll simply propagate that result upwards. This will make is so that if `onBatch` fails, then `onStore` will also failure, and hopefully the whole test will collapse at that point.

Reply:
ah, you're right, it's a closure 👍